homeassistant/components/actiontec/device_tracker.py

```diff
@@ -12,17 +12,13 @@
     DeviceScanner,
 )
 from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
 import homeassistant.helpers.config_validation as cv
-import homeassistant.util.dt as dt_util
+from homeassistant.helpers.typing import ConfigType
 
-_LOGGER = logging.getLogger(__name__)
+from .const import LEASES_REGEX, UNKNOWN_DEVICE_NAME, Device
 
-_LEASES_REGEX = re.compile(
-    r"(?P<ip>([0-9]{1,3}[\.]){3}[0-9]{1,3})"
-    + r"\smac:\s(?P<mac>([0-9a-f]{2}[:-]){5}([0-9a-f]{2}))"
-    + r"\svalid\sfor:\s(?P<timevalid>(-?\d+))"
-    + r"\ssec"
-)
+_LOGGER = logging.getLogger(__name__)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
Mark `PLATFORM_SCHEMA` in `homeassistant.const` as `Final`. Then see that this is overwriting core constant. Not good 🙂

Reply:
But it seems to be usual :see_no_evil: 
```
$ grep "PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend" homeassistant/components/*/*.py | wc -l
593
```

Reply:
@frenck how is this supposed to work? Is it ok that components modify core constants?

Reply:
ok ... description of `Schema.extend()` says
```
Create a new `Schema` by merging this and the provided `schema`.

        Neither this `Schema` nor the provided `schema` are modified. The
        resulting `Schema` inherits the `required` and `extra` parameters of
        this, unless overridden.
[...]
```

Reply:
But it's written back to `PLATFORM_SCHEMA` overwriting it.

Reply:
Intention of `PLATFORM_SCHEMA` is described here https://developers.home-assistant.io/docs/configuration_yaml_index/#platform_schema.
I'm not sure, if simple renaming `PLATFORM_SCHEMA` during import should solve the problem?! :thinking:

Reply:
I did some tests and it seems to work as expected and it doesn't modify `PLATFORM_SCHEMA` from `homeassistant.components.device_tracker`.

But this will modify it:
```python
from homeassistant.components import device_tracker
device_tracker.PLATFORM_SCHEMA = device_tracker.PLATFORM_SCHEMA.extend(...)
```

`mypy` will warn about assigning final name in both cases if `homeassistant.components.device_tracker.PLATFORM_SCHEMA` is `Final`.

So this code works, it's just not very good. I'd rather write this to make it cleaner:
```python
from homeassistant.components.device_tracker import PLATFORM_SCHEMA as BASE_PLATFORM_SCHEMA
```